src/main/webapp/js/common.js

```diff
@@ -484,20 +484,22 @@ function getPointValue(s, ditchZero) {
 /**
  * Checks if element is within browser's viewport.
  * @return true if it is within the viewport, false otherwise
+ * @see http://stackoverflow.com/q/123999
  */
 function isWithinView(element) {
-    var viewHeight = window.innerHeight;
-    var viewTop = window.scrollY;
-    var viewBottom = viewTop + viewHeight;
-    
-    var elementHeight = element.offsetHeight;
-    var elementTop = element.offsetTop;
-    var elementBottom = elementTop + elementHeight;
+    var baseElement = $(element)[0]; // unwrap jquery element
+    var rect = baseElement.getBoundingClientRect();
+
+    var windowHeight = window.innerHeight || document.documentElement.clientHeight;
+    var windowWidth = window.innerWidth || document.documentElement.clientWidth;
     
-    return viewHeight >= elementHeight
-           ? viewTop <= elementTop && viewBottom >= elementBottom          // all within view
-           : viewTop <= elementTop && viewBottom >= elementTop             // top within view
-             || viewTop <= elementBottom && viewBottom >= elementBottom;   // btm within view
+    var tolerance = 0.25;
```

Comment:
Add a comment about what the tolerance means, even with the stack overflow, it takes a deep reading in order to understand and not all maintainers will easily understand it

Also, it seems that the way we are using tolerance here is slightly different, the answer that it is taken from uses it as a percentage while we use it as an absolute pixel, can you elaborate on why we chose 0.25 pixels?